service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/DateFormatUtil.java

```diff
@@ -1,26 +1,48 @@
 package uk.nhs.adaptors.gp2gp.ehr.utils;
 
+import lombok.SneakyThrows;
+
+import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
-import java.time.format.DateTimeFormatterBuilder;
-import java.util.Date;
+
+import org.hl7.fhir.dstu3.model.DateTimeType;
+import org.hl7.fhir.dstu3.model.InstantType;
 
 import uk.nhs.adaptors.gp2gp.ehr.exception.EhrMapperException;
 
 public class DateFormatUtil {
-    private static final String UK_ZONE_ID = "Europe/London";
-    private static final DateTimeFormatter DATE_TIME_FORMATTER = new DateTimeFormatterBuilder()
-        .appendPattern("yyyyMMddHHmmss")
-        .toFormatter();
 
-    public static String formatDate(Date date) {
-        if (date == null) {
+    @SneakyThrows
+    public static String formatDateTimeType(DateTimeType dateTimeType) {
+        if (dateTimeType == null) {
```

Comment:
I think that the null should be checked be outside of the method. Passing null values (most of the time) is a bad practice.

Reply:
I don't agree, it is good that class gives a meaningful error instead of null pointer exception. This null will come here only in case of error.

Reply:
We can throw this exception outside of method, right before it is invoked.

Reply:
but then everybody who will ever use that public method with have to know about it.

Reply:
@Broom94 , are we planning to pass a null value here often?